Assets/MRTK/SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -118,6 +118,7 @@ public override bool IsInteractionEnabled
 
         private void Awake()
         {
+            PrioritizedLayerMasksOverride = GrabLayerMasks;
```

Comment:
I wonder if, instead of manually setting this here, we'd want to make `BaseControllerPointer`'s `PrioritizedLayerMasksOverride` property `virtual`, then override it in this class to point to our `grabLayerMasks` field (and deprecate the existing `GrabLayerMasks` property). As this line makes pretty obvious, we kinda have an existing duplicated concept here. Thoughts?

Reply:
that sounds like a good idea. I think the layer masks for pointers might need a retooling in general, but I'd rather tackle that in its own PR

Reply:
Are we sure that prioritized layer masks are configurable on a per-pointer basis? I remember we added the layer masks for the PokePointer and later the GrabPointer and for some reason did not use the PrioritizedLayerMasksOverride, though at the moment I cannot remember why. One thing I noticed is that the PrioritizedLayerMasksOverride property is not currently exposed as configurable in editor for any of the pointer types, like the GrabLayerMasks and PokeLayerMasks properties are. 

Being able to configure the grab layer masks in editor is something we want to keep because it exists and is being used by people already. @keveleigh if you wanted to deprecate grab layer masks, how do you envision users from editor POV configure the grabbable and poke-able layers? 

Additionally to both of you, do you think that the naming "grab layer masks" is a bit more understandable than "prioritized layer masks override"? While internally it does seem fine to use the PrioritizedLayerMasks property, perhaps exposing the easier-to-understand "GrabLayerMasks" property, and "PokeLayerMasks" property, is something we could still keep. 

So, I would suggest reconsidering whether to deprecate GrabLayerMasks in the future, perhaps we can just internally use the inherited PrioritizedLayerMasks property, but keep the user-facing API the same.

Reply:
@julenka To expand a bit on my proposal, I agree with you that the user-configurable field should definitely stick around. I was proposing to reconcile the read-only `GrabLayerMasks` property with the base class' `PrioritizedLayerMasks` property, instead of just setting them to be the same thing in this line. The serialized field currently backing `GrabLayerMasks` should definitely stay (so nothing will change as far as the inspector is concerned), and I think it should instead back `PrioritizedLayerMasksOverride` while the read-only property is deprecated.

As far as naming, I think it's fine to keep using the specifically named layer mask terminology in the inspector, but it feels odd to me to redefine an existing concept with a new name. I'd prefer consistency, personally, so that if you know how to set layer masks on one pointer you know how to set them on all, instead of looking for a specific pointer's custom defined layer mask property. It also lets you set them based on the pointer interface instead of the concrete type. Additionally, the existing property is read-only, while `PrioritizedLayerMasksOverride` can be set at runtime as well.

Going forward though, you bring up a good point that `PrioritizedLayerMasksOverride` isn't exposed in the inspector anywhere currently. It doesn't affect us here, since we already have a serialized field for layer masks in this class, but it might be interesting to expose at a lower layer at some point, so other pointer types have the option as well.

Reply:
Updated to deprecate out GrabLayerMasks